libp2p/connmanager.nim

```diff
@@ -396,7 +410,11 @@ proc storeConn*(c: ConnManager, conn: Connection)
     raise newException(LPError, "Connection closed or EOF")
 
   let peerId = conn.peerId
-  if c.conns.getOrDefault(peerId).len > c.maxConnsPerPeer:
+
+  if peerId in c.expectedConnections and
+     not(c.expectedConnections.getOrDefault(peerId).finished):
+      c.expectedConnections.getOrDefault(peerId).fut.complete(conn)
```

Comment:
Is `getOrDefault` necessary? Does `peerId in c.expectedConnections` imply always the presence of a value for the key?

Reply:
This is just to avoid the `KeyError` that `[]` can raise (even though it can't actually raise it since we checked)
Ideally we would have a `[]` that raises a defect if the value is not here to avoid this issue